sigstore-java/src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -37,34 +40,49 @@
 import java.security.SignatureException;
 import java.security.cert.CertificateException;
 import java.security.spec.InvalidKeySpecException;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import org.checkerframework.checker.nullness.qual.Nullable;
 
 /** A full sigstore keyless signing flow. */
-public class KeylessSigner {
+public class KeylessSigner implements AutoCloseable {
```

Comment:
Is autocloseable really necessary here? We're not holding any system resources or really anything of size. I assume GC at KeylessSigning cleanup is sufficient to release the memory used for the certs? Does this have to do with the gradle plugin signing service?

Reply:
Yeah, `AutoCloseable` is to release certificates (e.g. to avoid accidental reuse). It enables using keyless signer with Java's try-with-resources.